### ISSUE TITLE
Why does node UMLS:C0394007 have name "Cerebral Palsy"?

### ISSUE THREAD

**Issue by saramsey** [RTXteam/RTX-KG2#56]
Thanks to Will Byrd for this issue report.

In KG2.6.4 (and in KG2.6.3, and probably earlier versions?), the KG2 node with `id` field `UMLS:C0394007` has `name` property "Cerebral Palsy", as can be shown here:

```
match (n {id: 'UMLS:C0394007'}) return n.name;
```

<img width="686" alt="Screen Shot 2021-05-26 at 8 46 43 AM" src="https://user-images.githubusercontent.com/5562107/119690876-ec602980-bdfe-11eb-9901-4896edc9a38b.png">

But navigating to the URI for that node ([https://identifiers.org/umls:C0394007](https://identifiers.org/umls:C0394007)), we see that the official name for that node in UMLS is "Cerebral Palsy, Atonic":

<img width="753" alt="Screen Shot 2021-05-26 at 8 47 52 AM" src="https://user-images.githubusercontent.com/5562107/119691050-1a456e00-bdff-11eb-9996-f71710c78505.png">

I am wondering if official node names in UMLS are not visible to us with our current ingestion process, and if we are getting only the "alternative" node names.

**Comment by saramsey** [comment]
Another possible explanation to consider is that the SRI team used a different (older?) version of UMLS in building their KG than we did in KG2.6.4?

**Comment by saramsey** [comment]
Someone could reach out to Chris Bizon to ask what version of UMLS they used in building the SRI reference KG.

**Comment by ericawood** [comment]
Also, we are using an outdated version of UMLS. (We use 2020AA and 2021AA is out.)

**Comment by saramsey** [comment]
> Also, we are using an outdated version of UMLS. (We use 2020AA and 2021AA is out.)

Good point. For all I know, SRI could be using 2021AA in their reference KG build.

**Comment by saramsey** [comment]
Here is the text of Will Byrd's email on the matter (in case I missed some details that might provide a clue):

> Hi everyone!

> 
> tldr; we found an import bug in kg2.6.3 for UMLS concept names
> 
> 
> I was playing around with the new kg2.6.3 with Thi and John (Cc'd) on Thursday.
> 
> We were looking at the properties of a node corresponding to the CURIE 'UMLS:C0394007', which I'm sure is coming from SemmedDB.
> 
> Originally we thought there was some weird data import error, since we saw the 'name' property appear twice.  Turns out this was due to my having loaded both kg2.6.3 and the SRI SemmedDB KG at the same time.  mediKanren 2 was displaying both sets of properties (this was the behavior we were seeing, Thi and John).
> 
> That was a problem on my end.  However, we did notice what seems to be a problem in kg2.6.3.  Here are the properties for the node corresponding to 'UMLS:C0394007':
> 
> Welcome to Racket v8.0 [cs].
> > (require "common.rkt")
> loading configuration defaults: /Users/webyrd/github/mediKanren/medikanren2/config.defaults.scm
> loading configuration overrides: /Users/webyrd/github/mediKanren/medikanren2/config.scm
> > (run* (k v) (cprop "UMLS:C0394007" k v))
> '(("category" "biolink:Disease")
>   ("category_label" "disease")
>   ("deprecated" "False")
>   ("description"
>    "\"A form of non-spastic cerebral palsy with decreased muscle tone; noticeably \\\"floppy\\\" muscles with poor or no head control; A type of cerebral palsy characterized by decreased muscle tone.; A heterogeneous group of nonprogressive motor disorders caused by chronic brain injuries that originate in the prenatal period, perinatal period, or first few years of life. The four major subtypes are spastic, athetoid, ataxic, and mixed cerebral palsy, with spastic forms being the most common. The motor disorder may range from difficulties with fine motor control to severe spasticity (see MUSCLE SPASTICITY) in all limbs. Spastic diplegia (Little disease) is the most common subtype, and is characterized by spasticity that is more prominent in the legs than in the arms. Pathologically, this condition may be associated with LEUKOMALACIA, PERIVENTRICULAR. (From Dev Med Child Neurol 1998 Aug;40(8):520-7)\"")
>   ("id" "UMLS:C0394007")
>   ("iri" "https://identifiers.org/umls:C0394007")
>   ("name" "Cerebral Palsy")
>   ("provided_by" "identifiers_org_registry:umls")
>   ("update_date" "2017"))
> 
> Here, the name is "Cerebral Palsy".
> 
> However, the actual name of that UMLS concept is "Cerebral Palsy, Atonic" (for example, see http://linkedlifedata.com/resource/umls/id/C0394007).
> 
> The SRI Semmeddb KG returns the correct name, along with the underlying UMLS Semantic Type, which is also helpful:
> 
> Process scheme finished
> Welcome to Racket v8.0 [cs].
> > (require "common.rkt")
> loading configuration defaults: /Users/webyrd/github/mediKanren/medikanren2/config.defaults.scm
> loading configuration overrides: /Users/webyrd/github/mediKanren/medikanren2/config.scm
> > (run* (k v) (cprop "UMLS:C0394007" k v))
> '(("category" "disease_or_phenotypic_feature")
>   ("id" "UMLS:C0394007")
>   ("name" "Cerebral Palsy, Atonic")
>   ("umls_type" "(\"T047\")")
>   ("umls_type_label" "(\"Disease or Syndrome\")")
>   ("xrefs"
>    "(\"MESH:D002547\" \"MDR:10080116\" \"NCI:C116906\" \"SNOMEDCT_US:192958009\" \"RCD:F23y1\" \"NCI_NICHD:C116906\" \"CHV:0000032259\")"))
> 
> John suspects you are taking the concept name from the wrong field, since he says both "Cerebral Palsy, Atonic" and "Cerebral Palsy" are both associated with the concept.  We want the more specific name, however, since there are multiple types of Cerebral Palsy.
> 
> In fact, there are a number of distinct UMLS CURIES in kg2.6.3 that all have the name "Cerebral Palsy", which we are sure is not correct.
> 
> If it would be possible to bring in the UMLS Semantic Type as well, that would also be useful.
> 
> Thank you!
> 
> Cheers,
> 
> --Will

**Comment by ericawood** [comment]
Here are all of the files with information about `UMLS:C0394007` in import order with relevant information about them from `kg2lindsey.rtx.ai`:

Source | Source ID | Name | Definition
--|--|--|--
`umls-mdr.ttl` | [`MEDDRA:10080116`](https://bioportal.bioontology.org/ontologies/MEDDRA?p=classes&conceptid=10080116) | Hypotonic cerebral palsy | None
`umls-msh.ttl` | [`MESH:D002547`](https://bioportal.bioontology.org/ontologies/MESH?p=classes&conceptid=D002547) | Cerebral Palsy | A heterogeneous group of nonprogressive motor disorders caused by chronic brain injuries that originate in the prenatal period, perinatal period, or first few         years of life. The four major subtypes are spastic, athetoid, ataxic, and mixed cerebral palsy, with spastic forms being the most common. The motor disorder may range from difficul        ties with fine motor control to severe spasticity (see MUSCLE SPASTICITY) in all limbs. Spastic diplegia (Little disease) is the most common subtype, and is characterized by spastic        ity that is more prominent in the legs than in the arms. Pathologically, this condition may be associated with LEUKOMALACIA, PERIVENTRICULAR. (From Dev Med Child Neurol 1998 Aug;40(        8):520-7)
`umls-nci.ttl` | [`NCIT:C116906`](https://ncithesaurus.nci.nih.gov/ncitbrowser/ConceptReport.jsp?dictionary=NCI_Thesaurus&code=C116906&ns=ncit) | Hypotonic Cerebral Palsy | A type of cerebral palsy characterized by decreased muscle tone.
`umls-snomedct_us.ttl` | [`SNOMED:192958009`](https://bioportal.bioontology.org/ontologies/SNOMEDCT?p=classes&conceptid=192958009) | Hypotonic cerebral palsy | A form of non-spastic cerebral palsy with decreased muscle tone; noticeably \"floppy\" muscles with poor or no head control
`umls-mth.ttl` | [`UMLS:C0270742`](http://linkedlifedata.com/resource/umls/id/C0270742) | Athetoid cerebral palsy | None

The MESH based entry for this CUI has the longest description. Due to https://github.com/RTXteam/RTX-KG2/blob/e6275ee3413b19edda1ee42cf7d7bb14d3beaa22/kg2_util.py#L538-L546 `MESH:D002547`'s name will be chosen as the name for all CUI nodes based on it, which are as follows:
- `UMLS:C0007789`
- `UMLS:C0023882`
- `UMLS:C0154695`
- `UMLS:C0154697`
- `UMLS:C0154698`
- `UMLS:C0270742`
- `UMLS:C0270807`
- `UMLS:C0338596`
- `UMLS:C0394003`
- `UMLS:C0394007`
- `UMLS:C0553767`
- `UMLS:C0751024`
- `UMLS:C0751025`

**Comment by saramsey** [comment]
I wonder if we can write a script `get_cuis.py` that produces a "TTL" file of *all* UMLS CUIs, including just their name fields and semantic types. We could then load it *first* in multi_ont_to_json_kg.py, and we could change the merge order for the `name` field to not depend on the description field but "first takes precedence". The script `get_cuis.py` could extract, for each CUI, the preferred name, like this:

```
mysql> select STR from MRCONSO where CUI='C0394007' and LAT='ENG' and TS='P' and STT='PF';
+------------------------+
| STR                    |
+------------------------+
| Cerebral Palsy, Atonic |
+------------------------+
```

**Comment by saramsey** [comment]
of course, to get *all* the CUIs, we could do

```
select CUI, STR from MRCONSO where LAT='ENG' and TS='P' and STT='PF';
```

**Comment by ericawood** [comment]
Looks like there are some duplicates:

mysql> select CUI, STR from MRCONSO where LAT='ENG' and TS='P' and STT='PF' limit 50;
+----------+------------------------------------------------------------+
| CUI      | STR                                                        |
+----------+------------------------------------------------------------+
| C1411876 | Developmental arithmetic disorder                          |
| C0236825 | Developmental expressive writing disorder                  |
| C0920296 | Developmental reading disorder                             |
| C0236826 | Developmental expressive language disorder                 |
| C0236826 | Developmental expressive language disorder                 |
| C0011768 | Developmental receptive language disorder                  |
| C0009655 | Conduct disorder, group type                               |
| C0009656 | Conduct disorder, solitary aggressive type                 |
| C0009657 | Conduct disorder, undifferentiated type                    |
| C0035951 | Rumination disorder of infancy                             |
| C0236802 | Gender identity disorder of childhood                      |
| C0236802 | Gender identity disorder of childhood                      |
| C0236802 | Gender identity disorder of childhood                      |
| C0236802 | Gender identity disorder of childhood                      |
| C0014089 | Functional encopresis                                      |
| C0009090 | Cluttering                                                 |
| C0009090 | Cluttering                                                 |
| C0009090 | Cluttering                                                 |
| C0009090 | Cluttering                                                 |
| C0034748 | Reactive attachment disorder of infancy or early childhood |
| C0034748 | Reactive attachment disorder of infancy or early childhood |
| C0041651 | Uncomplicated alcohol withdrawal                           |
| C0236656 | Dementia associated with alcoholism                        |
| C0236669 | Cannabis delusional disorder                               |
| C0009176 | Cocaine intoxication                                       |
| C0009178 | Cocaine withdrawal                                         |
| C0236688 | Cocaine delirium                                           |
| C0236670 | Cocaine delusional disorder                                |
| C0236680 | Hallucinogen hallucinosis                                  |
| C0236671 | Hallucinogen delusional disorder                           |
| C0236702 | Hallucinogen mood disorder                                 |
| C0021452 | Inhalant intoxication                                      |
| C0028047 | Nicotine withdrawal                                        |
| C0028047 | Nicotine withdrawal                                        |
| C0029100 | Opioid intoxication                                        |
| C0029104 | Opioid withdrawal                                          |
| C0029225 | Organic delusional disorder                                |
| C0029232 | Organic mood disorder                                      |
| C0236748 | Organic anxiety disorder                                   |
| C0236748 | Organic anxiety disorder                                   |
| C0236748 | Organic anxiety disorder                                   |
| C0236748 | Organic anxiety disorder                                   |
| C0029233 | Organic personality disorder                               |
| C0029233 | Organic personality disorder                               |
| C0029233 | Organic personality disorder                               |
| C0018528 | Hallucinogen dependence                                    |
| C0018528 | Hallucinogen dependence                                    |
| C0018526 | Hallucinogen abuse                                         |
| C0021450 | Inhalant dependence                                        |
| C0029095 | Opioid abuse                                               |
+----------+------------------------------------------------------------+

**Comment by ericawood** [comment]
This appears to be fixed in KG2.6.5:
![image](https://user-images.githubusercontent.com/36611732/122796682-85098e00-d273-11eb-9989-9034b5507bec.png)
